volmdlr/wires.py

```diff
@@ -4335,57 +4346,76 @@ def close_sewing(self, dict_closing_pairs):
 
     def check_sewing(self, polygon2, sewing_faces):
         if not len(self.line_segments) + len(polygon2.line_segments) == len(sewing_faces):
-            print('len(self.line_segments) + len(polygon2.line_segments):',
-                  len(self.line_segments) + len(polygon2.line_segments))
-            print('len(sewing_faces):', len(sewing_faces))
             return False
         return True
 
+    def redefine_sewing_triangles_points(self, triangles_points,
+                                         passed_by_zero_index,
+                                         closing_point_index,
+                                         previous_closing_point_index):
+        for n, triangle_points in enumerate(triangles_points[::-1]):
+            if (not passed_by_zero_index and
+                self.points.index(
+                    triangle_points[2]) > closing_point_index) or \
+                    (passed_by_zero_index and
+                     0 <= self.points.index(triangle_points[
+                                                       2]) <= previous_closing_point_index and
+                     self.points.index(
+                         triangle_points[2]) > closing_point_index):
+                new_face_points = [triangles_points[-(n + 1)][0],
+                                   triangles_points[-(n + 1)][1],
+                                   self.points[
+                                       closing_point_index]]
+                triangles_points[-(n + 1)] = new_face_points
+
+        return triangles_points
+
+    @staticmethod
+    def clean_sewing_closing_pairs_dictionary(dict_closing_pairs,
+                                              closing_point_index,
+                                              passed_by_zero_index):
+        previous_closing_point_index = list(dict_closing_pairs.values())[-1][1]
+        for n in range(len(dict_closing_pairs)):
+            last_dict_value = list(dict_closing_pairs.values())[-1][1]
+            if (not passed_by_zero_index and
+                last_dict_value > closing_point_index) or\
+                    (passed_by_zero_index and
+                     0 <= last_dict_value <= previous_closing_point_index and
+                     last_dict_value > closing_point_index):
+                lower_bounddary_closing_point = \
+                            list(dict_closing_pairs.keys())[-1]
+                del dict_closing_pairs[
+                    list(dict_closing_pairs.keys())[-1]]
+
+        return dict_closing_pairs, lower_bounddary_closing_point
```

Comment:
I don't recommend: 
- turning a dict to a list, at least not in the iteration process, because a dict isn't ordered. You can convert it once to a list and then work on that new list. Then, converting it back to a dict, if you want.
- iterating on a list, from which you are deleting values.

Reply:
Done. Thank you